CorsixTH/Lua/entities/humanoids/vip.lua

```diff
@@ -202,268 +288,224 @@ function Vip:announce()
   end
 
-  -- evaluate the room we're currently looking at
-  for object, _ in pairs(room.objects) do
-    if object.object_type.id == "extinguisher" then
-      self.room_eval = self.room_eval + 1
-      break
-    elseif object.object_type.id == "plant" then
-      if object.days_left >= 10 then
-        self.room_eval = self.room_eval + 1
-      elseif object.days_left <= 3 then
-        self.room_eval = self.room_eval - 1
+--[[-- Group factor 2: Staff tiredness--]]
+  -- First get staff members
+  local count_staff = #self.hospital.staff
+  if count_staff > 1 then
+    -- Loop through staff tiredness, if any above verytired, break loop
+    for _, staff in ipairs(self.hospital.staff) do
+      if staff.attributes["fatigue"] ~= nil and staff.attributes["fatigue"] >= 0.7 then
+        self.vip_rating = self.vip_rating + 2
+        break
       end
-      break
     end
-
-    if object.strength then
-      if object.strength > (object.object_type.default_strength / 2) then
-        self.room_eval = self.room_eval + 1
-      else
-        self.room_eval = self.room_eval - 3
-      end
+    -- Average all staff tiredness. If above tiredness level award 1, else -1
+    local avg_tired = self.hospital:getAverageStaffAttribute("fatigue", 1)
+    if avg_tired >= 0.5 then
+      self.vip_rating = self.vip_rating + 1
+    else
+      self.vip_rating = self.vip_rating - 1
     end
+  else
+    -- Penalise where there is one or no staff
+    self.vip_rating = self.vip_rating + 4
   end
-end
 
-function Vip:evaluateEmergency(success)
-  -- Make sure that the VIP is still actually in the process of evaluation
-  if not self.going_home then
-    if success then
-      self.vip_rating = self.vip_rating + 10
+--[[-- Group factor 3: Patients--]]
+  -- first check we had patients this visit
+  local patients_this_visit = self.enter_patients + self.hospital.num_visitors - self.enter_visitors
+  if patients_this_visit > 0 then
+    -- Average all patient health
+    local avg_health = self.hospital:getAveragePatientAttribute("health", 0.19)
+    if avg_health >= 0.2 then
+      self.vip_rating = self.vip_rating - 1
     else
-      self.vip_rating = self.vip_rating - 15
+      self.vip_rating = self.vip_rating + 1
     end
-  end
-end
 
-function Vip:setVIPRating()
-  --check the visitor to patient death ratio
-  local death_diff = self.hospital.num_deaths - self.enter_deaths
-  local visitors_diff = self.hospital.num_visitors - self.enter_visitors
-  if death_diff == 0 then
-    if visitors_diff ~= 0 then --if there have been no new patients, no +/- points
-      self.vip_rating = self.vip_rating + 20
+    -- get patient warmth
+    local avg_warmth = self.hospital:getAveragePatientAttribute("warmth", nil)
+    -- punish if too cold/hot
+    if avg_warmth then
+      local patients_warmth_ratio_rangemap = {
+        {upper = 0.21, value = 2},
+        {upper = 0.36, value = -1},
+        {upper = 0.43, value = 1},
+        {value = 2}
```

Comment:
Do you technically need another lower range value here, ie 0.21 is only 0.5 degress less than the comfort range, but if 3 degrees (0.42) hotter than comfortable, its penalised half as much, or just remove the 0.43 range?